plugins/notifications-backend-module-email/src/processor/NotificationsEmailProcessor.ts

```diff
@@ -215,20 +217,50 @@ export class NotificationsEmailProcessor implements NotificationProcessor {
     );
   }
 
-  private async sendPlainEmail(notification: Notification, emails: string[]) {
+  private getNotificationLink(notification: Notification) {
+    if (notification.payload.link) {
+      const stripLeadingSlash = (s: string) => s.replace(/^\//, '');
+      const ensureTrailingSlash = (s: string) => s.replace(/\/?$/, '/');
+
+      try {
+        const url = new URL(
+          stripLeadingSlash(notification.payload.link),
+          ensureTrailingSlash(this.frontendBaseUrl),
+        );
+        return url.toString();
+      } catch (_e) {
+        // noop: fallback to relative URL
+      }
+      return notification.payload.link;
+    }
+    return `${this.frontendBaseUrl}/notifications`;
+  }
+
+  private getHtmlContent(notification: Notification) {
     const contentParts: string[] = [];
     if (notification.payload.description) {
       contentParts.push(`${notification.payload.description}`);
     }
-    if (notification.payload.link) {
-      contentParts.push(`${notification.payload.link}`);
+    const link = this.getNotificationLink(notification);
+    contentParts.push(`<a href="${link}">${link}</a>`);
```

Comment:
i wonder if there's a way to inject malicious code here ... the link is unencoded; in theory you could both render a `javascript:` link as well as something containing like `"></a><script>...` 🤔 Note for the future: look into escaping or sanitizing the contents here

Reply:
Thanks for pointing that out, makes sense. I actually moved this to `notifications-node` as it is needed also in the #24599 so a little bit of refactoring needed here anyways

Reply:
Actually, this should also be checked in the notifications-backend when a notification link is created for the first time. Have to talk more with @mareklibra about this when he is back.